### PR TITLE
Dataverse 6142 solr schema changes

### PR DESCRIPTION
**Pull request by donsizemore**

### CODE REVIEW

**Comment by donsizemore** [comment]
@pdurbin there are a number of points in the role that won't work in "every" version, but the 80/20 rule would probably focus on develop/release. i had a question surrounding @poikilotherm's use of the shell script versus a pre-fab schema.xml(?) but... I just touched up the branch a little bit and ran with it.

**Comment by donsizemore** [comment]
@poikilotherm no. the ansible role may run against any branch (if release, it uses dvinstall.zip). the Dataverse version may be specified as a group_var, but without an archive of artifacts tied to each release by version number, the choices are effectively develop/branch or release.

**Comment by pdurbin** [comment]
@donsizemore as of 713a57f I just tried the following:

./ec2-create-instance.sh -g ec2config.yaml -a dataverse-6142-solr-schema-changes

I was trying to spin up the "develop" branch.

Here's my group vars file: [ec2config.yaml.txt](https://github.com/IQSS/dataverse-ansible/files/3633156/ec2config.yaml.txt)

I got the following errors:

TASK [dataverse : release: place solr schema] **********************************
skipping: [localhost] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [dataverse : branch: place solr schema files] *****************************
fatal: [localhost]: FAILED! => {"msg": "The task includes an option with an undefined variable. The error was: 'dataserse' is undefined\n\nThe error appears to have been in '/home/centos/dataverse/tasks/dataverse-solr.yml': line 68, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n- name: \"branch: place solr schema files\"\n  ^ here\nThis one looks easy to fix.  It seems that there is a value started\nwith a quote, and the YAML parser is expecting to see the line ended\nwith the same kind of quote.  For instance:\n\n    when: \"ok\" in result.stdout\n\nCould be written as:\n\n   when: '\"ok\" in result.stdout'\n\nOr equivalently:\n\n   when: \"'ok' in result.stdout\"\n"}
	to retry, use: --limit @/home/centos/dataverse/dataverse.retry

PLAY RECAP *********************************************************************
localhost                  : ok=68   changed=49   unreachable=0    failed=1

**Comment by pdurbin** [comment]
@donsizemore I moved this out of code review but I'm happy to help troubleshoot. Maybe a YAML formatting error? I'm not sure.

**Comment by poikilotherm** [comment]
Pretty sure var `dataserse` has a typo in it :smile:

File: tasks/dataverse-solr.yml [hunk 1]
```diff
@@ -55,47 +55,28 @@
...
-        src: "{{ dataverse.srcdir }}/conf/solr/{{ dataverse.solr.version }}/schema.xml"
-        dest: "{{ dataverse.solr.root }}/server/solr/collection1/conf/" 
-        owner: "{{ dataverse.solr.user }}"
-        group: "{{ dataverse.solr.group }}"
-        mode: 0644
-        remote_src: yes
+- name: "branch: place solr schema files"
+  copy:
+    src: '{{ dataverse.srcdir }}/conf/solr/schema_*.xml'
+    dest: '{{ dataserse.solr.root }}/server/solr/collection1/conf'
```
[the remark this answers is not in the document]
Comment (pdurbin) [reply tasks/dataverse-solr.yml]: @poikilotherm good catch! Created pull request #105 and testing it now.

**Comment by pdurbin** [comment]
Works great! I just tested "develop" and "release". Thanks, @donsizemore !!

**Pull request merged**